xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -909,6 +909,11 @@ void CAESinkALSA::Deinitialize()
 
 void CAESinkALSA::Stop()
 {
+  m_controlMonitor.Clear();
```

Comment:
Deinitialze is called quite often by AE, not only on exit

Reply:
I could move this into the destructor.
But Clear() is already called everytime in EnumerateDevicesEx:
https://github.com/a1rwulf/xbmc/blob/8f81b6a9056639daaafe0a57c3fcc600b204136a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp#L1165
It will then restart the monitor some lines below.

Reply:
I just wanted to point out that Stop i not the right place for the reason given. I don't know what clear does.

Reply:
I was pointed towards this patch to try after filing a bug report at: https://trac.kodi.tv/ticket/17793#ticket
I haven't experienced any more segfaults when quitting Kodi since applying it so this patch seems to have fixed the bug I reported. I don't know if this will be the final version that's merged but I thought I'd share my experience.

Reply:
Thx for your report.
Most likely it stops monitoring too often and will be revisited.

Reply:
@a1rwulf what was the reason to not add it into the dtor instead?

Reply:
Just need to do it - until know I don't know a reason why it should not work when I move it into dtor.
Will update in an hour or so.